components/brave_wallet/browser/internal/hd_key.h

```diff
@@ -92,27 +87,35 @@ class HDKey : public HDKeyBase {
                                int recid);
 
  private:
+  FRIEND_TEST_ALL_PREFIXES(Eip1559TransactionUnitTest,
+                           GetSignedTransactionAndHash);
+  FRIEND_TEST_ALL_PREFIXES(Eip2930TransactionUnitTest,
+                           GetSignedTransactionAndHash);
+  FRIEND_TEST_ALL_PREFIXES(EthereumKeyringUnitTest, SignMessage);
+  FRIEND_TEST_ALL_PREFIXES(EthTransactionUnitTest, GetSignedTransactionAndHash);
   FRIEND_TEST_ALL_PREFIXES(HDKeyUnitTest, GenerateFromExtendedKey);
   FRIEND_TEST_ALL_PREFIXES(HDKeyUnitTest, SetPrivateKey);
   FRIEND_TEST_ALL_PREFIXES(HDKeyUnitTest, SetPublicKey);
   FRIEND_TEST_ALL_PREFIXES(HDKeyUnitTest, SignAndVerifyAndRecover);
 
+  // value must be 32 bytes
+  void SetPrivateKey(base::span<const uint8_t> value);
+  void SetChainCode(base::span<const uint8_t> value);
+  // value must be 33 bytes valid public key (compressed)
+  void SetPublicKey(const std::vector<uint8_t>& value);
+
   void GeneratePublicKey();
   const std::vector<uint8_t> Hash160(const std::vector<uint8_t>& input);
-  std::string Serialize(
-      uint32_t version,
-      std::unique_ptr<std::vector<uint8_t>, SecureZeroVectorDeleter<uint8_t>>
-          key) const;
-
-  uint8_t depth_;
-  uint32_t fingerprint_;
-  uint32_t parent_fingerprint_;
-  uint32_t index_;
+  std::string Serialize(uint32_t version, base::span<const uint8_t> key) const;
+
+  uint8_t depth_ = 0;
+  uint32_t fingerprint_ = 0;
+  uint32_t parent_fingerprint_ = 0;
+  uint32_t index_ = 0;
   std::vector<uint8_t> identifier_;
-  std::unique_ptr<std::vector<uint8_t>, SecureZeroVectorDeleter<uint8_t>>
-      private_key_;
+  SecureVector private_key_;
   std::vector<uint8_t> public_key_;
-  std::vector<uint8_t> chain_code_;
+  SecureVector chain_code_;
```

Comment:
This should be also secured as public key + chain code is enough to deconstruct key subtree.